### ISSUE TITLE
Missing wrapper for config.enable_device

### ISSUE THREAD

**Issue by rambo** [jerry73204/realsense-rust#6]
Makes it rather hard to handle multiple devices in a deterministic manner.

**Comment by rambo** [comment]
edit: Made some changes...

~Someone see https://github.com/rambo/realsense-rust/commit/0f051244b000a4ff4b2cbd44f1094214bc03a892 and tell me how to fix that TODO.~ WIP at https://github.com/advian-oss/realsense-rust/tree/issue-6

Rust test:

```rust
use failure::Fallible;
use realsense_rust;


#[tokio::main]
async fn main() -> Fallible<()> {
    let ctx = realsense_rust::Context::new()?;
    let devices = ctx.query_devices(None)?;
    let dcount = devices.len().unwrap();
    if dcount == 0 {
        return Err(failure::err_msg("No RS devices found"));
    }

    let mut device_configs: Vec<realsense_rust::Config> = Vec::with_capacity(dcount);
    for device in devices {
        let device = device.unwrap();
        let name = device.name().unwrap().unwrap();
        let sn = device.serial_number().unwrap().unwrap();
        println!("Found {} SN {}", name, sn);
        let config = realsense_rust::Config::new()?.enable_device(sn)?;
        let dtype = device.product_line().unwrap().unwrap();
        if dtype == "D400" {
            let config = config.enable_stream(realsense_rust::StreamKind::Color, 0, 1280,720,realsense_rust::Format::Bgr8, 30)?;
            let config = config.enable_stream(realsense_rust::StreamKind::Depth, 0, 1280,720,realsense_rust::Format::Z16, 30)?;
            device_configs.push(config);
        }
        else if dtype == "T200" {
            let config = config.enable_stream(StreamKind::Pose, 0, 0, 0, RsFormat::_6Dof, 200)?;
            device_configs.push(config);
        }
        else {
            println!("Don't know what to do with {} devices", dtype);
        }
    }

    for config in device_configs {
        println!("====== ======");
        let pipeline = realsense_rust::Pipeline::new()?;
        let mut pipeline = pipeline.start_async(Some(config)).await?;

        // show stream info
        let profile = pipeline.profile();
        for (idx, stream_result) in profile.streams()?.try_into_iter()?.enumerate() {
            let stream = stream_result?;
            println!("stream data {}: {:#?}", idx, stream.get_data()?);
        }
    }

    Ok(())
}
```

With D400 and T200 device connected this finds the D400 streams fine but T200 pipeline init says "Error: RealSense error: No device connected" both on OSX and arm64 Linux.

**Comment by rambo** [comment]
ref T265 problems, I am apparently blind
"""T265 is currently supported on Linux and Windows. Preliminary Mac OS support is now available. Android support is planned but not yet included.""" no wonder it's not always perfect

edit: not that it works any better on arm64 linux with T265, D400 devices work fine on both. I need to make a (better, old one at https://gist.github.com/rambo/485d7f41892b8a9a9814aa6db2a9a4da) minimal C++ test and run it on Linux to see if the problem is actually on Intels end on the actually-supported platform.

**Comment by jerry73204** [comment]
I released 0.3.2 and it ships [enable_device_from_{serial,file}](https://docs.rs/realsense-rust/0.3.2/realsense_rust/config/struct.Config.html#method.enable_device_from_serial) methods. Can you try it out?

**Comment by jerry73204** [comment]
The feature is published on crates.io. Now the issue can be closed.
